src/object/cli_obj.c

```diff
@@ -5107,28 +5129,53 @@ obj_retry_next_shard(struct dc_object *obj, struct obj_auxi_args *obj_auxi,
 static inline bool
 need_retry_redundancy(struct obj_auxi_args *obj_auxi)
 {
+	/* NB: If new failure being added here, then please update failure check in
+	 * obj_shard_comp_cb() as well.
+	 */
 	return (obj_auxi->csum_retry || obj_auxi->tx_uncertain || obj_auxi->nvme_io_err);
 }
 
+/* Check if the shard was failed in the previous fetch, so these shards can be skipped */
+static inline bool
+shard_was_fail(struct obj_auxi_args *obj_auxi, uint32_t shard_idx)
+{
+	struct obj_auxi_tgt_list *failed_list;
+	uint32_t tgt_id;
+
+	if (obj_auxi->force_degraded) {
+		D_DEBUG(DB_IO, DF_OID" fail idx %u\n",
+			DP_OID(obj_auxi->obj->cob_md.omd_id), shard_idx);
+		obj_auxi->force_degraded = 0;
```

Comment:
seems not must to clear "obj_auxi->force_degraded" flag?
now DIOF_FOR_FORCE_DEGRADE flag only used for rebuild code and will set obj_auxi->force_degraded if carried  DIOF_FOR_FORCE_DEGRADE flag